File: 206.ReverseLinkedList/reverse.py
# Definition for singly-linked list.
# class ListNode(object):
#     def __init__(self, x):
#         self.val = x
#         self.next = None

class Solution(object):
    def reverseList(self, head):
        """
        :type head: ListNode
        :rtype: ListNode
        """
        newHead = None
        curr = head
        while curr:
            prev = curr
            curr = curr.next
            prev.next = newHead
            newHead = prev
        return newHead

File: 206.ReverseLinkedList/reverse.cc
/**
 * Definition for singly-linked list.
 * struct ListNode {
 *     int val;
 *     ListNode *next;
 *     ListNode(int x) : val(x), next(NULL) {}
 * };
 */
class Solution {
public:
    ListNode* reverseList(ListNode* head) {
        ListNode *newHead = NULL, *curr = head;
        while (curr) {
            ListNode *prev = curr;
            curr = curr->next;
            prev->next = newHead;
            newHead = prev;
        }

        return newHead;
    }
};
